src/zoid/buttons/util.js

```diff
@@ -106,6 +108,21 @@ export function getVenmoExperiment(experiment : ?Experiment) : VenmoExperiment {
     };
 }
 
+export function getButtonsArray(props : ButtonProps) : $ReadOnlyArray<$Values<typeof FUNDING>> {
```

Comment:
I'm concerned about seeing the `getVenmoExperiment` call in this `getButtonsArray` function. Can you explain the purpose of this function? I'd like to see if we can simplify the logic.

Reply:
I went through the other [SPB PR](https://github.com/paypal/paypal-smart-payment-buttons/pull/209) and it seems like we want to know if there is only one button rendered or if there are more than one. Which buttons are actually rendered doesn't seem to matter. And we need to know this information on second render so it can be passed to the personalization api call. Is that correct?

Reply:
Hi Greg! So, I created this `getButtonsArray` function to get the buttons array and know if the page will render single or multiple buttons, according to the array length. And about the `getVenmoExperiment`, is because this prop is mandatory to call the `determineEligibleFunding` function, which returns the buttons array.

Reply:
About the second comment, you are right! I don't use the buttons array to know which button will be rendered, but how many will be rendered. And with this information, I can assign a boolean to my variable `hasMultipleButtons`, which will be sent to the Personalization graphql call. In general, I'm using the same logic as we already use on [this part](https://github.com/paypal/paypal-checkout-components/blob/d408ae23346f7ce93797e06d1f15078592b94bff/src/ui/buttons/buttons.jsx#L104)

Reply:
Thanks for explaining. That makes a lot more sense now. True button eligibility is a combination of the embedded server-side eligibility plus the client-side eligibility rules (support for shipping callback, tagline, layout, etc). This is unknown to SPB right now so I can understand why this new query param prop is needed.